ui/app/models/pki/pki-role-engine.js

```diff
@@ -6,25 +6,76 @@ import { withModelValidations } from 'vault/decorators/model-validations';
 import fieldToAttrs from 'vault/utils/field-to-attrs';
 
 const validations = {
-  name: [
-    { type: 'presence', message: 'Name is required.' },
-    {
-      type: 'containsWhiteSpace',
-      message: 'Name cannot contain whitespace.',
-    },
-  ],
+  name: [{ type: 'presence', message: 'Name is required.' }],
 };
 
 @withModelValidations(validations)
 export default class PkiRoleEngineModel extends Model {
   @attr('string', { readOnly: true }) backend;
+
   @attr('string', {
     label: 'Role name',
-    fieldValue: 'id',
-    readOnly: true,
+    fieldValue: 'name',
   })
   name;
 
+  @attr('string', {
+    label: 'Issuer Reference',
+    subText:
+      'Specifies the default issuer of this request. May be the value default, a name, or an issuer ID. To find this, you can view the issuer or run: read -field=default <mount-name>/config/issuers in the CLI.',
+  })
+  issuerRef;
+
+  @attr({
+    label: 'Not valid after',
+    subText:
+      'The time after which this certificate will no longer be valid. This can be a TTL (a range of time from now) or a specific date. If not set, the system uses "default" or the value of max_ttl, whichever is shorter. Alternatively, you can set the not_after date below.',
+    editType: 'yield',
+  })
+  customTtl;
+
+  @attr({
+    label: 'Backdate validity',
+    subText:
+      'Also called the notBeforeDuration property. Allows certificates to be valid for a certain time period before now. This is useful to correct clock misalignment on various systems when setting up your CA.',
+    editType: 'ttl',
+    hideToggle: true,
```

Comment:
Yep, you're right. the default on the api is 30s. I'll go ahead and change.

Reply:
Oh, and sorry, by default this is what is shows. Unless you specifically change it to not show a default of 30s, it will editType ttl will default to 30s. 
![image](https://user-images.githubusercontent.com/6618863/192842255-a1c8b710-a643-4411-90ab-463235f39e40.png)

Reply:
(the label is Backdate validity but the param is `notBeforeDuration`

Reply:
And just noticing the subText does not show on this field... I'll fix that.

Reply:
Done. 
![image](https://user-images.githubusercontent.com/6618863/192842771-b0f59c28-48de-4194-85f6-cf5379144b0f.png)

Reply:
If you look at the payload on submit, you'll notice that `not_before_duration` is sent as the wrong type. it's submitted as `30` not `30s` (this is reflected in the ember data inspector, and then once you edit the input, it changes to a string)

<img width="200" alt="Screen Shot 2022-09-28 at 3 31 04 PM" src="https://user-images.githubusercontent.com/68122737/192900051-02cd5123-e8e0-4d9c-9999-c784e5db02d6.png">